integr8ly/ansible-tower-configuration
# Creation of repository comparison playbook

**Summary**
If a new variables is added to the Tower Dummy Credentials project, it should also be added to the Integreatly Dev project to ensure that both projects are in sync i.e that they contain the same variables.

This PR contains the following additions:

1. Bootstrapping of newly created tower instances to include the required resources to run a repository comparison
2. A related playbook that compares both repositories (compare_credential_repositories.yml)

If differences exist, i.e, the Tower Dummy Credentials project does contains variables that have not been added to integreatly dev, an email will be sent to the CI/CD team (Robert O' Brien. Paul McCarthy and Steven Tobin) to notify them of this so action can be taken to fix the issue.

**Linked Pull Requests**
https://github.com/fheng/integreatly_dev/pull/19

**Prerequisites**

1. A tower instance to run the bootstrap process against
2. Access to the Integreatly Dev repository

**Validation Steps**

1. Checkout this feature branch
2. Bootstrap your tower instance (as per: https://github.com/integr8ly/ansible-tower-configuration#4-bootstrapping), and also include the following enviromental variables and values in the bootstrapping command (using `-e <variable-name>=<variable_value>`)

- `tower_configuration_project_scm_type_url`: `https://github.com/obrienrobert/ansible-tower-configuration.git`

- `tower_configuration_project_scm_branch`: `credential_repository_sync`

- `cluster_credential_git_ref`: `send_grid_credentials`

3. Run a new `Compare Credential Repositories` job from your tower instance. As no differences exist between both projects, the job should run successfully and no email should be send to the users.
